vendor/bat-native-ledger/src/bat/ledger/internal/database/database_activity_info.cc

```diff
@@ -166,11 +167,23 @@ void DatabaseActivityInfo::NormalizeList(
 
   transaction->commands.push_back(std::move(command));
 
-  auto transaction_callback = std::bind(&OnResultCallback,
-      _1,
-      callback);
+  auto shared_list = std::make_shared<ledger::PublisherInfoList>(
+      std::move(list));
+
+  ledger_->ledger_client()->RunDBTransaction(
+      std::move(transaction),
+      [this, shared_list, callback](ledger::DBCommandResponsePtr response) {
+        if (!response || response->status !=
+              ledger::DBCommandResponse::Status::RESPONSE_OK) {
+          callback(ledger::Result::LEDGER_ERROR);
+          return;
+        }
+
+        ledger_->ledger_client()->PublisherListNormalized(
```

Comment:
It looks like this is a fix to make sure that `PublisherListNormalized` is called only after the database is updated. Is that correct?

Reply:
yes as values will only change if db is successfully updated and only in this case we need to normalize based on new values